dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/ResourceLink.java

```diff
@@ -0,0 +1,165 @@
+                    resourceLink.append(HTTPS_PREFIX);
+                }else{
+                    resourceLink.append(HTTP_PREFIX);
+                }
+                resourceLink.append(host.getHostname());
+                if(request.getServerPort() != 80 && request.getServerPort() != 443){
+                    resourceLink.append(StringPool.COLON).append(request.getServerPort());
+                }
+                resourceLinkUri.append(identifier.getParentPath()).append(contentlet.getStringProperty(FileAssetAPI.FILE_NAME_FIELD));
+                resourceLink.append(UtilMethods.encodeURIComponent(resourceLinkUri.toString()));
+                resourceLinkUri.append(LANG_ID_PARAM).append(contentlet.getLanguageId());
+                resourceLink.append(LANG_ID_PARAM).append(contentlet.getLanguageId());
+            }
+
+            final FileAsset fileAsset = getFileAsset(contentlet);
+            final String mimeType = fileAsset.getMimeType();
+            final String fileAssetName = fileAsset.getFileName();
+
+            return new ResourceLink(resourceLink.toString(), resourceLinkUri.toString(), mimeType, fileAsset, isEditableAsText(mimeType, fileAssetName), isDownloadRestricted(fileAssetName));
+        }
+
+        private static boolean isEditableAsText(final String mimeType, final String fileAssetName ){
+             return  (
+                 !isRestrictedMimeType(mimeType) && (isEditableMimeType(mimeType) || fileAssetName.endsWith(".vm"))
+             );
+        }
+
+        private static boolean isRestrictedMimeType(final String mimeType){
+           return (mimeType.contains("officedocument") || mimeType.contains("svg"));
+        }
+
+        private static boolean isEditableMimeType(final String mimeType) {
+            return (
+                    mimeType.contains("text") ||
+                    mimeType.contains("javascript") ||
+                    mimeType.contains("json") ||
+                    mimeType.contains("xml") ||
+                    mimeType.contains("php")
+            );
+        }
+
+        Host getHost(final String hostId, final User user) throws DotDataException, DotSecurityException{
+            return APILocator.getHostAPI().find(hostId , user, false);
+        }
+
+        Identifier getIdentifier(final Contentlet contentlet) throws DotDataException {
+            return APILocator.getIdentifierAPI().find(contentlet);
+        }
+
+        FileAsset getFileAsset(final Contentlet contentlet){
+           return APILocator.getFileAssetAPI().fromContentlet(contentlet);
+        }
+    }
+
+    /**
+     * This method is used to determined if file the extension should be allowed for download or not.
+     * @param fileAssetName
+     * @return
+     */
+    public static boolean isDownloadRestricted(final String fileAssetName ){
+        final String lowerCaseAssetName = fileAssetName.toLowerCase();
+        return lowerCaseAssetName.endsWith(".vm") || lowerCaseAssetName.endsWith(".vtl");
```

Comment:
it would be good to have these extensions as a constants or even an util such as isVelocityFile